lib/features/thread/presentation/thread_controller.dart

```diff
@@ -3,57 +3,162 @@ import 'package:core/presentation/utils/responsive_utils.dart';
+  final loadMoreState = LoadMoreState.IDLE.obs;
+
+  late Set<Comparator> sortCurrent = Set();
+  late Filter filterCurrent;
+  late AccountId accountId;
+  int positionCurrent = ThreadConstants.POSITION_DEFAULT;
+  int limitCurrent = ThreadConstants.LIMIT_DEFAULT;
+
+  ThreadController(this.responsiveUtils, this._getAllEmailInteractor);
+
+  void updateMailbox({required PresentationMailbox presentationMailbox, required AccountId accountId}) {
+    this.accountId = accountId;
+    this.mailboxCurrent.value = presentationMailbox;
+    this.selectedEmail.value = PresentationEmail.presentationEmailEmpty;
+
+    filterCurrent = mailboxCurrent.value.id != PresentationMailbox.presentationMailboxEmpty.id
+        ? EmailFilterCondition(inMailbox: mailboxCurrent.value.id)
+        : EmailFilterCondition();
+
+    refreshGetAllEmailAction();
+  }
+
+  @override
+  void onData(Either<Failure, Success> newState) {
+    super.onData(newState);
+  }
+
+  @override
+  void onDone() {
+    viewState.value.fold(
+      (failure) {
+        if (loadMoreState.value == LoadMoreState.LOADING) {
+          positionCurrent -= limitCurrent;
+        }
+        loadMoreState.value = LoadMoreState.COMPLETED;
+      },
+      (success) {
+        if (success is GetAllEmailSuccess && success.emailList.isNotEmpty) {
+          emailList.value += success.emailList;
+          loadMoreState.value = LoadMoreState.IDLE;
+          print('SIZE: ${emailList.length}');
+        } else {
+          if (loadMoreState.value == LoadMoreState.LOADING) positionCurrent -= limitCurrent;
+          loadMoreState.value = LoadMoreState.COMPLETED;
+        }
+      }
+    );
+  }
 
   @override
-  void onReady() {
-    super.onReady();
-    mailboxDashBoardController.scaffoldKey.currentState?.openEndDrawer();
+  void onError(error) {
+    if (loadMoreState.value == LoadMoreState.LOADING) {
+      positionCurrent -= limitCurrent;
+    }
+    loadMoreState.value = LoadMoreState.COMPLETED;
+  }
+
+  void getAllEmailAction(AccountId accountId,
+    {
+      int limit = ThreadConstants.LIMIT_DEFAULT,
+      int position = ThreadConstants.POSITION_DEFAULT,
+      Set<Comparator>? sort,
+      Filter? filter
+    }
+  ) async {
+    consumeState(_getAllEmailInteractor.execute(
+      accountId,
+      limit: limit,
+      position: position,
+      sort: sort,
+      filter: filter,
+      properties: Email.allProperties
+    ));
   }
 
-  SelectMode getSelectMode(PresentationThread presentationThread) {
-    return presentationMailCurrent.value.id == presentationThread.id
+  void refreshGetAllEmailAction() {
+    loadMoreState.value = LoadMoreState.IDLE;
+    positionCurrent = 0;
+    emailList.value = [];
+
+    getAllEmailAction(
+      accountId,
+      position: positionCurrent,
+      limit: limitCurrent,
+      sort: sortCurrent,
+      filter: filterCurrent);
+  }
+
+  void loadMoreEmailAction() {
+    loadMoreState.value = LoadMoreState.LOADING;
+    positionCurrent += limitCurrent;
```

Comment:
You should get the next emails start from the position which is 
`positionCurrent += lastGetTotal`